packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1041,6 +1043,23 @@ abstract class ParentDataWidget<T extends RenderObjectWidget> extends ProxyWidge
     return result;
   }
 
+  /// Write the data from this widget into the given render object's parent data.
+  ///
+  /// The framework calls this function whenever it detects that the
+  /// [RenderObject] associated with the [child] has outdated
+  /// [RenderObject.parentData]. For example, if the render object was recently
+  /// inserted into the render tree, the render object's parent data might not
+  /// match the data in this widget.
+  ///
+  /// Subclasses are expected to override this function to copy data from their
+  /// fields into the [RenderObject.parentData] field of the given render
+  /// object. The render object's parent is guartuneed to have been created by a
+  /// widget of type `T`, which usually means the render object's parent data
```

Comment:
i think this falls off the rails after "which usually". Maybe "which usually means the widget knows what kind of parent data object it will be updating" or something?